src/ast.ls

```diff
@@ -1001,15 +1001,26 @@ class exports.Call extends Node
             ++index
         node <<< back: (args[index] = fun)body
 
-    @let = (args, body, generator = false) ->
+    @let = (args, body) ->
+        has-yield = false
+        has-await = false
+        body.traverse-children (child) ->
+            if child instanceof Yield
+                switch child.op
+                | \yield \yieldfrom => has-yield := true
+                | \await            => has-await := true
+            return true if has-yield and has-await
```

Comment:
A function can never legally have both a yield and an await, so this is a little misleading. Could you change this back to something a little closer to the original expression in `add-body`, where you return as soon as any instance of `Yield` is found? Instead of returning true you can return `child.op`, to distinguish between the yield and await cases without mutating a variable in the outer scope.

Reply:
This was left intentionally for async generators, as I would imagine they will come to LS eventually. I thought it would be better to future-proof it, in case someone implementing async generators forgets about `async`s and `yield`s in `let` blocks, as it have happened with async functions.

Reply:
That's a good thought! I wasn't aware that async generators had gotten this far in the standardization process before now; they should be very easy to add to LiveScript, so I agree this is a good precaution.